gi-docgen.yaml

```diff
@@ -15,49 +27,84 @@ environment:
       - busybox
       - ca-certificates-bundle
       - meson
-      - py3-jinja2
-      - py3-markdown
-      - py3-markupsafe
-      - py3-packaging
-      - py3-pygments
-      - py3-typogrify
-      - python3
+      - py3-supported-jinja2
+      - py3-supported-markdown
+      - py3-supported-markupsafe
+      - py3-supported-packaging
+      - py3-supported-pip
+      - py3-supported-pygments
+      - py3-supported-python
+      - py3-supported-typogrify
 
 pipeline:
   - uses: fetch
     with:
       expected-sha256: c2f49f0bdea0366d04b6ab6f34071f324ef70016bc3650371cc08d3c6f2764da
       uri: https://gitlab.gnome.org/GNOME/gi-docgen/-/archive/${{package.version}}/gi-docgen-${{package.version}}.tar.gz
 
-  - runs: |
-      meson . output
-      meson compile ${JOBS:+-j ${JOBS}} -C output
-      DESTDIR="${{targets.destdir}}" meson install --no-rebuild -C output
-
-  - uses: strip
-
 subpackages:
-  - name: py3-gi-docgen
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
     dependencies:
+      provider-priority: ${{range.value}}
+      provides:
+        - py3-${{vars.pypi-package}}
+        - gi-docgen
       runtime:
-        - py3-markdown
-        - py3-markupsafe
-        - py3-pygments
-        - py3-jinja2
-        - py3-packaging
-        - py3-typogrify
-        - py3-tomli
-        - python3
+        - py${{range.key}}-jinja2
+        - py${{range.key}}-markdown
+        - py${{range.key}}-markupsafe
+        - py${{range.key}}-packaging
+        - py${{range.key}}-pygments
+        - py${{range.key}}-typogrify
+        - python-${{range.key}}
     pipeline:
-      - name: Python Build
-        uses: python/build-wheel
+      - runs: |
+          py=python${{range.key}}
+          pyfullpath=$(which "$py") || {
+            echo "ERROR: '$py' not found in PATH."
+            exit 1
+          }
+
+          echo "Using Python interpreter: $pyfullpath"
+
+          # Create a temporary directory and symlink for python3
+          tmp_dir=$(mktemp -d)
+          trap "rm -rf $tmp_dir" EXIT
+          ln -s "$pyfullpath" "$tmp_dir/python3"
+          export PATH="$tmp_dir:$PATH"
```

Comment:
meson build required python3 in the PATH, with `py3-supported-python` package it is not set with these we create a temp dir have a symlink of python3 for each version, add it in the PATH, complete the meson build
and cleanup the temp dir using `trap "rm -rf $tmp_dir" EXIT` at the end.